### PR TITLE
[ROCm] Remove expected inductor UT fails for batch norm

### PR DESCRIPTION
**Pull request by jataylo**
Removing expected failures relating to inductor batch_norm on ROCm

Also removing the addition of `tanh` to expected failures list as this is a cuda exclusive failure already captured here (cc: @peterbell10)
```
if not TEST_WITH_ROCM:
    inductor_gradient_expected_failures_single_sample["cuda"]["tanh"] = {f16}
```

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Review by peterbell10 — APPROVED** [review]

**Comment by jataylo** [comment]
Will rebase and merge

**Comment by jithunnair-amd** [comment]
ROCm CI failure is unrelated to this PR. Merging

**Comment by huydhn** [comment]
@pytorchbot drci

(Please ignore this comment, I'm only testing dr.ci)
